ArcGISRuntimeSDKQt_CppSamples/Scenes/AnimateImagesWithImageOverlay/AnimateImagesWithImageOverlay.cpp

```diff
@@ -0,0 +1,204 @@
+  QObject(parent),
+  m_scene(new Scene(this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // create a new tiled layer from World_Dark_Gray_Base REST service
+  ArcGISTiledLayer* worldDarkGrayBasemap = new ArcGISTiledLayer(QUrl("https://services.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+
+  // add the tiled layer to the operational layers to display the basemap
+  m_scene->operationalLayers()->append(worldDarkGrayBasemap);
+}
+
+AnimateImagesWithImageOverlay::~AnimateImagesWithImageOverlay() = default;
+
+void AnimateImagesWithImageOverlay::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<AnimateImagesWithImageOverlay>("Esri.Samples", 1, 0, "AnimateImagesWithImageOverlaySample");
+}
+
+SceneQuickView* AnimateImagesWithImageOverlay::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void AnimateImagesWithImageOverlay::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  emit sceneViewChanged();
+
+  setupImageOverlay();
+}
+
+void AnimateImagesWithImageOverlay::setupImageOverlay()
+{
+  // create an envelope of the pacific southwest sector for displaying the image frame
+  const Point pointForImageFrame(-120.0724273439448, 35.131016955536694, SpatialReference(4326));
+  m_pacificSouthwestEnvelope = Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+
+  // create a camera, looking at the pacific southwest sector
+  const Point observationPoint(-116.621, 24.7773, 856977);
+  const Camera camera(observationPoint, 353.994, 48.5495, 0);
+  const Viewpoint pacificSouthwestViewpoint(observationPoint, camera);
+
+  m_scene->setInitialViewpoint(pacificSouthwestViewpoint);
+
+  // create an image overlay
+  m_imageOverlay = new ImageOverlay(this);
+
+  // append the image overlay to the scene view
+  m_sceneView->imageOverlays()->append(m_imageOverlay);
+
+  // Create new Timer and set the timeout interval to 68
+  // 68 ms interval timer equates to approximatey 15 frames a second
+  m_timer = new QTimer(this);
+  m_timer->setInterval(68);
+
+  // connect to the timeout signal to load and display a new image frame each time
+  connect(m_timer, &QTimer::timeout, this, &AnimateImagesWithImageOverlay::animateImageFrames);
+}
+
+void AnimateImagesWithImageOverlay::animateImageFrames()
+{
+  const QString dataPath = defaultDataPath() + "/ArcGIS/Runtime/Data/3D/ImageOverlay/PacificSouthWest";
```

Comment:
Seems like this could be stored as a member to avoid the call to defaultDataPath() every time you animate. Anything in this method will be executed several times per second so we should make sure it is absolutely required.